panther_analysis_tool/test_case.py

```diff
@@ -35,21 +35,28 @@ def __init__(self, data: Dict[str, Any], data_model: Optional[DataModel]) -> Non
         """
         self._data = data
         self.data_model = data_model
+        self._case_insensitive_key_map: Dict[str, str] = {}
+        # Support case-insensitive lookups for top-level fields only
+        for key in self._data:
+            self._case_insensitive_key_map.setdefault(key.lower(), key)
 
     def __getitem__(self, arg: str) -> Any:
-        return self._data.get(arg, None)
```

Comment:
I took the liberty of modifying the behavior here as mentioned before, which also uncovered a small bug in a `panther-analysis` test.